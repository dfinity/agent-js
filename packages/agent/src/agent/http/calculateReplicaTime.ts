/**
 * Parse the expiry from the message
 * @param message an error message
 * @returns diff in milliseconds
 */
export const calculateReplicaTime = (message: string): Date => {
  const [min, max] = message.split('UTC');

  const minsplit = min.trim().split(' ').reverse();

  const minDateString = `${minsplit[1]} ${minsplit[0]} UTC`;

  const maxsplit = max.trim().split(' ').reverse();

  const maxDateString = `${maxsplit[1]} ${maxsplit[0]} UTC`;

  return new Date(minDateString);
};

function midwayBetweenDates(date1: Date, date2: Date) {
  return new Date((date1.getTime() + date2.getTime()) / 2);
}
