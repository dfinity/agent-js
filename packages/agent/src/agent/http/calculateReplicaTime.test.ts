import { calculateReplicaTime } from './calculateReplicaTime';
const exampleMessage = `Specified ingress_expiry not within expected range: Minimum allowed expiry: 2024-08-13 22:49:30.148075776 UTC, Maximum allowed expiry: 2024-08-13 22:55:00.148075776 UTC, Provided expiry:        2021-01-01 00:04:00 UTC`;

test('calculateReplicaTime', () => {
  const parsedTime = calculateReplicaTime(exampleMessage);
  expect(parsedTime).toMatchSnapshot();
});
