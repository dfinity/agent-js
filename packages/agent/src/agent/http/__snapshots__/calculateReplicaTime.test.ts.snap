// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateReplicaTime 1`] = `2024-08-13T22:49:30.148Z`;
